lib/kubeclient_next/client/rest_client.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+require "faraday"
+require "base64"
+
+module KubeclientNext
+  module Client
+    class RESTClient
+      attr_reader :group, :version
+
+      def initialize(config:, context:, group:, version:)
+        @config = config
+        @context = context
+        @group = group
+        @version = version
+
+        cert_store = OpenSSL::X509::Store.new
+        cert_store.add_cert(OpenSSL::X509::Certificate.new(Base64.decode64(cluster.certificate_authority_data)))
+        @connection = Faraday.new(
+          endpoint,
+          ssl: {
+            client_cert: OpenSSL::X509::Certificate.new(Base64.decode64(user.auth_info.client_certificate_data)),
+            client_key: OpenSSL::PKey::RSA.new(Base64.decode64(user.auth_info.client_key_data)),
+            cert_store: cert_store,
+          }
+        )
```

Comment:
Again, more hardcoding, of a sort. Future work will need to take care of the various ways in which auth can be performed. I'm almost tempted to write a Faraday middleware module to take care of this, but it probably doesn't matter too much, either way. The important thing to keep in mind here is to maintain a mechanism for automatic auth renewal (in the case of bearer tokens, JWTs, etc.)